CHANGELOG.md

```diff
@@ -1,5 +1,46 @@
 # Changelog
 
+## [1.2](https://github.com/arborx/arborx/tree/v1.2) (2022-04-01)
+
+[Full Changelog](https://github.com/arborx/arborx/compare/v1.1...v1.2)
+
+**New features:**
+
+- Implement Euclidean minimum spanning tree algorithm [\#624](https://github.com/arborx/ArborX/pull/624), [\#631](https://github.com/arborx/ArborX/pull/631), [\#661](https://github.com/arborx/ArborX/pull/631)
```

Comment:
Should we say it it experimental?  It is defined in `ArborX::Details::` which usually signify you are not allowed to use it.

Reply:
I think it would be better fit in `namespace Experimental` than `namespace Details` then.

Reply:
I would not worry about what namespace it live in right now.